File: code/1018/1018.rb
n = gets.to_i

def print_and_return_calc(quant, value)
  puts "#{quant / value} nota(s) de R$ #{value},00"
  quant - (quant / value) * value
end

puts n
n = print_and_return_calc(n, 100)
n = print_and_return_calc(n, 50)
n = print_and_return_calc(n, 20)
n = print_and_return_calc(n, 10)
n = print_and_return_calc(n, 5)
n = print_and_return_calc(n, 2)
print_and_return_calc(n, 1)

File: code/1018/1018.js
const input = require('fs').readFileSync('/dev/stdin', 'utf8');
const lines = input.split('\n');

const printAndCalcBanknotes = (quant, value) => {
  const actualQuant = Math.floor(quant / value);
  console.log(`${actualQuant} nota(s) de R\$ ${value},00`);
  return quant - (actualQuant * value);
}

let n = parseInt(lines[0]);

console.log(n);

n = printAndCalcBanknotes(n, 100);
n = printAndCalcBanknotes(n, 50);
n = printAndCalcBanknotes(n, 20);
n = printAndCalcBanknotes(n, 10);
n = printAndCalcBanknotes(n, 5);
n = printAndCalcBanknotes(n, 2);
printAndCalcBanknotes(n, 1);
